### DISCUSSION TITLE
"CreatedBy" property cannot be added to the Filter Webpart

### DISCUSSION THREAD

**Discussion by ValiPrk** [microsoft-search/pnp-modern-search#3367] [General]
We are using Modern PNP Search - 4.8.0 on our SharePoint Online environment and would like to add as a refiner in the Filter Webpart the "CreatedBy" property.
I know that by default the property is not refinable, I tried to map it under a "RefinableString" property, but it will not be available to be added on the Search Results webpart first and after on the Filter Webpart.
Also I tried to use the "Author" property, but it sees also the last user that edited the document as Author and it's confusing.
Is there a way to make it work with the CreatedBy property?

**Comment by kasperbolarsen** [comment]
Hi @ValiPrk 
Absolutely :-) First map RefinableStringXX to these crawled properties
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/8f3217ad-b272-4071-b7c9-141e957d917b)

Once you can see data begining to show up in the RefinableString ( using e.g. SP Editor or SharePoint Query Tool) you can add RefinableStringXX to Selected Properties in the Results web part, and finally add RefinableStringXX as a filter in the Search Filters web part:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ddcb479f-c18e-42f3-bb01-3233916889d7)

**Reply by ValiPrk** [reply]
Hi @kasperbolarsen 

Thank you for your solution, it works, but is there a way to display only the DisplayName, not the whole value like e-mail, DisplayName and so on?

**Reply by kasperbolarsen** [reply]
I am not sure I understand the question.  What do you see?

**Reply by ValiPrk** [reply]
I'm seeing results like on the 2'nd value from your screenshot: e-mail address | DisplayName | GID.... 
![results](https://github.com/microsoft-search/pnp-modern-search/assets/147488275/42aeddbc-ee30-4695-8789-3827684f31cf)

**Comment by kasperbolarsen** [comment]
I have been looking into it tonight. The definition for CreatedBy (Office:#4, AUTHOR, MAILFROM) seems to be too broad for our purpose. The question is what you mean by "CreatedBy" for a news story in the initial request? Is it the person that created the page initially? Is it the person that modified the page most resently ( most likely not) OR is it the name displayed on the page:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/f3ab5858-b02f-4fea-ad2f-45fd13ad0acb)

I have worked a lot with intranets some years back, and we often saw that one person in the Comms depart made the news story and added the relevant Department Head as the AuthorByline ( the person shown on the story). If you are looking for this person you should check if the crawled property OWS_AUTHORBYLINE is mapped to a RefinableString, and map it if no mapping exists.

**Reply by ValiPrk** [reply]
Hi @kasperbolarsen,

I want to display the creator of the content (document, news page, etc), but right now I have another issue, all my names are displayed like numbers (see attached screenshot), did you ever encounter such a issue in the past? 
![Numbers](https://github.com/microsoft-search/pnp-modern-search/assets/147488275/41933ae9-88fa-4e66-bfc1-69e4565851af)

**Reply by ValiPrk** [reply]
Hi @kasperbolarsen, 

I managed to resolve the issue, when I mapped the "Office:#4" property, it showed me exactly what I needed.
Thank you for your help and wish you merry holidays!
Valentin

![Correct mapping](https://github.com/microsoft-search/pnp-modern-search/assets/147488275/2ef3ad1f-5d0b-47d1-b4e5-61bcdc2198ae)
